gammapy/data/tests/test_data_store.py

```diff
@@ -91,22 +95,63 @@ def test_datastore_subset(tmpdir, data_manager):
 @requires_dependency('yaml')
 def test_data_summary(data_manager):
     """Test data summary function"""
-    
+
     data_store = data_manager['hess-crab4-hd-hap-prod2']
     t = data_store.data_summary([23523, 23592])
     assert t[0]['events'] == 620975
     assert t[1]['edisp_2d'] == 28931
 
     t = data_store.data_summary([23523, 23592], summed=True)
     assert t[0]['psf_3gauss'] == 6042
+
+
+@pytest.mark.parametrize("pars,result", [(dict(energy=None, theta=None),
+                                          dict(energy_shape=18, theta_shape=300, psf_energy=2.5178505859375 * u.TeV,
+                                               psf_theta=0.05 * u.deg,
+                                               psf_exposure=Quantity(6878545291473.34, "cm2 s"),
+                                               psf_value=Quantity(205215.42446175334, "1/sr"))),
+                                         (dict(energy=EnergyBounds.equal_log_spacing(1, 10, 100, "TeV"), theta=None),
+                                          dict(energy_shape=101, theta_shape=300, psf_energy=1.2589254117941673 * u.TeV,
+                                               psf_theta=0.05 * u.deg,
+                                               psf_exposure=Quantity(4622187644084.735, "cm2 s"),
+                                               psf_value=Quantity(119662.71915415104, "1/sr"))),
+                                         (dict(energy=None, theta=Angle(np.arange(0, 2, 0.002), 'deg')),
+                                          dict(energy_shape=18, theta_shape=1000, psf_energy=2.5178505859375 * u.TeV,
+                                               psf_theta=0.02 * u.deg,
+                                               psf_exposure=Quantity(6878545291473.34, "cm2 s"),
+                                               psf_value=Quantity(23082.369133891403, "1/sr"))),
+                                         (dict(energy=EnergyBounds.equal_log_spacing(1, 10, 100, "TeV"),
+                                               theta=Angle(np.arange(0, 2, 0.002), 'deg')),
+                                          dict(energy_shape=101, theta_shape=1000,
+                                               psf_energy=1.2589254117941673 * u.TeV,
+                                               psf_theta=0.02 * u.deg,
+                                               psf_exposure=Quantity(4622187644084.735, "cm2 s"),
+                                               psf_value=Quantity(27987.773313506143, "1/sr"))),
+                                         ])
+@requires_data('gammapy-extra')
+def test_make_psf(pars, result):
+    center = SkyCoord(83.63, 22.01, unit='deg')
+    store = gammapy_extra.filename("datasets/hess-crab4-hd-hap-prod2")
+    data_store = DataStore.from_dir(store)
+    obs1 = data_store.obs(23523)
+    psf = obs1.make_psf(source_position=center, energy=pars["energy"], theta=pars["theta"])
+    assert_allclose(len(psf.offset), result["theta_shape"])
```

Comment:
Can you please change all these `len` function calls to `.shape` attribute calls. I.e. instead of `len(psf.offset)` call `psf.offset.shape` and check that.

This is a better check, because `len` of a multi-dimensional array will return the length of the first (or last, not sure) dimension, and checking the shape explicitly tests more.